src/lib/math.js

```diff
@@ -326,8 +326,40 @@ const $builtinmodule = function (name) {
         }
     };
 
-    function isqrt(x) {
-        throw new Sk.builtin.NotImplementedError("math.isqrt() is not yet implemented in Skulpt");
+    function bit_length(i) {
+        return new Sk.builtin.int_(Sk.builtin.bin(i).sq$length() - 2);
+    }
+
+    function isqrt(n) {
+        n_js = Sk.misceval.asIndexOrThrow(n);
+        n = new Sk.builtin.int_(n_js);
```

Comment:
I'm not sure I like this approach of working with Python objects for `isqrt`.
It's seems a little obtuse.

What about working exclusively with javascript objects?
If `n_js` is a number we can just do the quick thing with `Math.` functions.

And if it's not a number we know it's a JSBI.BigInt.
we can then follow a sqrt algorithm using `JSBI.` calls.

Reply:
I've had a go at doing this. Let me know of any suggested improvements (especially for the BigInt part).

Cheers,
Jeff.